bot.py

```diff
@@ -122,155 +177,365 @@ def run_script(self):
+                            resized_full_card_image = cv2.resize(
+                                zoomed_card_image, standard_size
+                            )
+
+                            similarity = self.image_processor.calculate_similarity(
+                                resized_api_card_image, resized_full_card_image
+                            )
+                            similarities.append((card, similarity))
+
+                        similarities.sort(key=lambda x: x[1], reverse=True)
+                        selected_card = similarities[0][0]
+                        self.log_callback(
+                            f"Highest similarity is {similarities[0][1]:.2f}"
+                        )
+
+                        # Show options to user
+                        event = threading.Event()
+                        self.ui_instance.show_card_options(
+                            similarities, zoomed_card_image, event
+                        )
+                        event.wait()
+                        selected_card = self.ui_instance.selected_card
+                        break
+                self.log_callback(f"Selected card: {selected_card}")
+                # Convert API data to our format
+                card_info = self.convert_api_card_data(selected_card)
+
+                # Update deck info with converted data
+                card_name = selected_card["name"].lower()
+                self.deck_info[selected_card["id"]] = card_info
+                self.card_images[selected_card["id"]] = zoomed_card_image
+                # write to the images/cards folder
+                cv2.imwrite(
+                    f"images/cards/{selected_card['id']}.png", zoomed_card_image
+                )
+                # Update all can_evolve flags
+                for key, value in self.deck_info.items():
+                    # Check if current card can evolve from any existing cards
+                    if value.get("evolves_from", "").lower() == card_name.lower():
+                        self.deck_info[key]["can_evolve"] = True
+
+                    # Check if any existing cards can evolve from current card
+                    if (
+                        card_info.get("evolves_from", "").lower()
+                        == value.get("name", "").lower()
+                    ):
+                        card_info["can_evolve"] = True
+
+                save_deck(self.deck_info)
+
             hand_cards.append(card_name.capitalize() if card_name else "Unknown Card")
 
-            card_info = self.deck_info.get(card_name, default_pokemon_stats)
-            
+            card_info = self.deck_info.get(card_name, None)
+            if card_info is None:
+                self.log_callback(
+                    f"Card {card_name} not found in deck, getting from api..."
+                )
+                # get from the api the card info
+                card = self.card_data_manager.get_card_by_id(card_name)
```

Comment:
Seems like it should be 
```suggestion
                card = self.card_data_manager.get_card_by_name(card_name)
```

Reply:
I let. id because now the images save with ID names, and on this case we already have the image, so we automatically get the card_name as the id instead of the name on the case where we have the image but not the data. Is a little confusing for now, but i think we can save all imagens and data as IDs to avoid conflicts.

Reply:
What kind of crashes you have?